### PR TITLE
[feat] Style and Tweets

### LINKED ISSUE #4: User actions
Implement ways to remove mentors and a button to open their profiles in a new tab.

### PR DESCRIPTION
**Pull request by shivampandey0**
## Style
- Imported [Marvel UI](https://marvel-ui.vercel.app/) for styling
- changed CSS class naming pattern
- Tweets Styling
- Searchbox Style
- removed the `Add` button in support of `Enter` to add

## Tweets Label
- Configured URL to exclude `replies` and differentiate between `Retweet` and `Tweet`
- Added badge icons to visually identify the Tweel and Retweet

and Fixes #4 

image

### CODE REVIEW

File: functions/tweets-fetch/tweets-fetch.js [hunk 1]
```diff
@@ -1,43 +1,39 @@
...
-    } catch (error) {
-        // output to netlify function log
-        console.log(error)
-        return {
-            statusCode: 500,
-            // Could be a custom message or object i.e. JSON.stringify(err)
-            body: JSON.stringify({ msg: error.message }),
-        }
+    if (!response.ok) {
+      // NOT res.status >= 200 && res.status < 300
```
Comment (Hamdrive) [inline functions/tweets-fetch/tweets-fetch.js]: Cleanup commented code

File: functions/tweets-fetch/tweets-fetch.js [hunk 2]
```diff
@@ -1,43 +1,39 @@
...
+    const data = await response.json();
+
+    return {
+      statusCode: 200,
+      body: JSON.stringify({ msg: data }),
+    };
+  } catch (error) {
+    return {
+      statusCode: 500,
+      // Could be a custom message or object i.e. JSON.stringify(err)
```
Comment (Hamdrive) [inline functions/tweets-fetch/tweets-fetch.js]: commented code here as well

File: src/components/Avatar.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
+const Avatar = ({ image, name = "Twitter Mentors" }) => {
```
Comment (Hamdrive) [inline src/components/Avatar.js:1]: Why the default parameter value?
Is this for if the API does not return a user profile name

File: src/components/Tweets.js [hunk 4]
```diff
@@ -1,41 +1,68 @@
...
+  const [tweets, setTweets] = useState([]);
+
+  // useEffect(() => {
+  //   async function getData() {
+  //     await fetch(`/.netlify/functions/tweets-fetch?user_id=${id}`)
+  //       .then((x) => x.json())
+  //       .then(handleTweets);
+  //   }
+  //   getData();
+  // }, [id]);
```
Comment (Hamdrive) [inline src/components/Tweets.js]: here as well

**Review by Hamdrive — COMMENTED** [review]
Good code concepts followed, better practices with React implemented.
Nice use of Component Library making creation of website much better

File: src/components/Avatar.js (hunk 3 above)
Comment (shivampandey0) [reply src/components/Avatar.js:1]: This `Avatar` component is shared between the `Sidebar` and `Tweets` section. 
On the Sidebar top, It shows the app name and on the right Tweets section, it gets the user name.

File: functions/tweets-fetch/tweets-fetch.js (hunk 1 above)
Comment (shivampandey0) [reply functions/tweets-fetch/tweets-fetch.js]: will do

File: functions/tweets-fetch/tweets-fetch.js (hunk 2 above)
Comment (shivampandey0) [reply functions/tweets-fetch/tweets-fetch.js]: will do.

File: src/components/Tweets.js (hunk 4 above)
Comment (shivampandey0) [reply src/components/Tweets.js]: will do

File: src/components/Avatar.js (hunk 3 above)
Comment (Hamdrive) [reply src/components/Avatar.js:1]: Understood, great logical thinking!

**Pull request merged**